src/util/secure_socket_enclave.c

```diff
@@ -254,33 +241,62 @@ ATTESTATION_STATUS generate_response(secure_message_t* req_message,
         return INVALID_PARAMETER_ERROR;
     }
 
-    // Call Pelz request message handler
-    ret = handle_pelz_request_msg((char*)decrypted_data, decrypted_data_length, &resp_data, &resp_data_length);
-    SAFE_FREE(decrypted_data);
-    if(ret)
+    // Store plaintext in session object
+    if(session_info->request_data != NULL) {
+        SAFE_FREE(session_info->request_data);
+    }
+
+    session_info->request_data = (char *) decrypted_data;
+    session_info->request_data_length = decrypted_data_length;
+
+    return SUCCESS;
+}
+
+//Construct an outgoing message containing data stored in the session object
+ATTESTATION_STATUS handle_outgoing_msg(size_t max_payload_size,
+                                       secure_message_t **resp_message,
+                                       size_t *resp_message_size,
+                                       size_t resp_message_max_size,
+                                       uint32_t session_id)
+{
+    char* resp_data;
+    size_t resp_data_length;
+    const uint8_t* plaintext;
+    uint32_t plaintext_length;
+    size_t resp_message_calc_size;
+    dh_session_t *session_info;
+    secure_message_t* temp_resp_message;
+    sgx_status_t status;
+
+    plaintext = (const uint8_t*)(" ");
```

Comment:
What is ```plaintext``` used for? I see it gets handed to ```sgx_rijndael128GCM_encrypt``` but I can't see where it's populated with useful data before then.

Reply:
`plaintext` is used as the "additional authentication data" argument to `sgx_rijndael128GCM_encrypt`, documented in https://github.com/intel/linux-sgx/blob/master/common/inc/sgx_tcrypto.h#L307

It is intentionally left empty here with size 0, so it has no effect on the encryption.

The variable name `plaintext` comes from the original LocalAttestation code.
If it seems too confusing, we could rename it, add a comment, or change it to NULL.

Reply:
I think either a rename to ```aad``` (the usual name for additional authenticated data) or a comment indicating it's not mean to be the plaintext that's getting encrypted, it's the additional authenticated data that GCM can accept.

Reply:
Because it's always empty and not actually being used for additional authenticated data, I think calling it `aad` may still cause confusion, so I made the argument NULL in [a98104e](https://github.com/NationalSecurityAgency/pelz/pull/121/commits/a98104e5568a18247a4d1b4f67519af57c1cb668).